NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -34,78 +52,224 @@ function Base.show(io::IO, s::CategoryProduct)
 ×(c1::AbstractCategory, c2::AbstractCategory) = ×(CategoryProduct(c1), CategoryProduct(c2))
 function ×(p1::CategoryProduct, p2::CategoryProduct)
   return CategoryProduct(categories_product(categories(p1), categories(p2)))
 end
 
-categories_product(l1::NamedTuple, l2::NamedTuple) = union_keys(l1, l2)
-
-categories_product(l1::Tuple, l2::Tuple) = (l1..., l2...)
-
+×(a, g::AbstractUnitRange) = ×(to_graded_axis(a), g)
+×(g::AbstractUnitRange, b) = ×(g, to_graded_axis(b))
 ×(nt1::NamedTuple, nt2::NamedTuple) = ×(CategoryProduct(nt1), CategoryProduct(nt2))
 ×(c1::NamedTuple, c2::AbstractCategory) = ×(CategoryProduct(c1), CategoryProduct(c2))
 ×(c1::AbstractCategory, c2::NamedTuple) = ×(CategoryProduct(c1), CategoryProduct(c2))
 
-#
-# Dictionary-like implementation
-#
+function ×(l1::LabelledInteger, l2::LabelledInteger)
+  c3 = label(l1) × label(l2)
+  m3 = unlabel(l1) * unlabel(l2)
+  return labelled(m3, c3)
+end
+
+function ×(g1::AbstractUnitRange, g2::AbstractUnitRange)
+  v = map(
+    ((l1, l2),) -> l1 × l2,
+    Iterators.flatten((Iterators.product(blocklengths(g1), blocklengths(g2)),),),
+  )
+  return gradedrange(v)
+end
+
+# ====================================  Fusion rules  ======================================
+# generic case: fusion returns a GradedAxes, even for fusion with Empty
+function fusion_rule(::SymmetryStyle, s1::CategoryProduct, s2::CategoryProduct)
+  return to_graded_axis(categories_fusion_rule(categories(s1), categories(s2)))
+end
+
+# Abelian case: fusion returns CategoryProduct
+function fusion_rule(::AbelianGroup, s1::CategoryProduct, s2::CategoryProduct)
+  return categories_fusion_rule(categories(s1), categories(s2))
+end
+
+# Empty case
+function fusion_rule(::EmptyCategory, ::EmptyCategoryProduct, ::EmptyCategoryProduct)
+  return sector()
+end
+
+# EmptyCategory acts as trivial on any AbstractCategory, not just CategoryProduct
+function fusion_rule(::SymmetryStyle, ::EmptyCategoryProduct, c::AbstractCategory)
+  return to_graded_axis(c)
+end
+function fusion_rule(::SymmetryStyle, ::EmptyCategoryProduct, c::CategoryProduct)
+  return to_graded_axis(c)
+end
+function fusion_rule(::SymmetryStyle, c::AbstractCategory, ::EmptyCategoryProduct)
+  return to_graded_axis(c)
+end
+function fusion_rule(::SymmetryStyle, c::CategoryProduct, ::EmptyCategoryProduct)
+  return to_graded_axis(c)
+end
 
+# abelian case: return Category
+fusion_rule(::AbelianGroup, ::EmptyCategoryProduct, c::AbstractCategory) = c
+fusion_rule(::AbelianGroup, ::EmptyCategoryProduct, c::CategoryProduct) = c
+fusion_rule(::AbelianGroup, c::AbstractCategory, ::EmptyCategoryProduct) = c
+fusion_rule(::AbelianGroup, c::CategoryProduct, ::EmptyCategoryProduct) = c
+
+# ===============================  Ordered implementation  =================================
+CategoryProduct(t::Tuple) = _CategoryProduct(t)
+CategoryProduct(cats::AbstractCategory...) = CategoryProduct((cats...,))
```

Comment:
```suggestion
CategoryProduct(cats::AbstractCategory...) = CategoryProduct(cats)
```
Julia implicitly makes `cats` a Tuple.